src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Tov.cpp

```diff
@@ -127,46 +194,87 @@ TovSolution::TovSolution(
 double TovSolution::outer_radius() const noexcept { return outer_radius_; }
 
 double TovSolution::mass(const double r) const noexcept {
+  ASSERT(r >= 0.0 and r <= outer_radius_, "Invalid radius: " << r << "\n");
   return mass_interpolant_(r);
 }
 
-Scalar<DataVector> TovSolution::mass(const Scalar<DataVector>& radius) const
-    noexcept {
-  DataVector mass(radius.size(), 0.0);
-  for (size_t i = 0; i < radius.size(); i++) {
-    mass[i] = mass_interpolant_(get(radius)[i]);
-  }
-  return Scalar<DataVector>{std::move(mass)};
-}
-
 double TovSolution::log_specific_enthalpy(const double r) const noexcept {
+  ASSERT(r >= 0.0 and r <= outer_radius_, "Invalid radius: " << r << "\n");
   return log_enthalpy_interpolant_(r);
 }
 
-Scalar<DataVector> TovSolution::log_specific_enthalpy(
-    const Scalar<DataVector>& radius) const noexcept {
-  DataVector log_specific_enthalpy(radius.size(), 0.0);
-  for (size_t i = 0; i < radius.size(); i++) {
-    log_specific_enthalpy[i] = log_enthalpy_interpolant_(get(radius)[i]);
+template <>
+RelativisticEuler::Solutions::TovStar<TovSolution>::RadialVariables<double>
+TovSolution::radial_variables(
+    const EquationsOfState::EquationOfState<true, 1>& equation_of_state,
+    const tnsr::I<double, 3>& x) const noexcept {
+  // add small number to avoid FPEs at origin
+  const double radius = get(magnitude(x)) + 1.e-30;
```

Comment:
Should this maybe be something like `1.e-30 * outer_radius_` to make it scale with the size of the star?